Quick Auto Clean Updates

MacSplody opened the pull request.

MacSplody committed:
Quick Auto Clean Updates

Common

Add entries for Quick Auto Clean Dirty Messages:

- QuickClean should be used when no deleted `nav` are found after QAC.
- reqManualFix should be used when deleted navmeshes are still present after QAC.

Update Cleaning Guide Link
- Now Links to Tome of xEdit.

Updated Cleaning Info for:
- DLC
- Unofficial Fallout 4 Patch

Freso commented on masterlist.yaml:
```diff
@@ -254,7 +254,13 @@ common:
+    info: 'A guide for running **Quick Auto Clean** can be found [here](https://tes5edit.github.io/docs/5-mod-cleaning-and-error-checking.html#quickautoclean).'
+  - &reqManualFix
+    util: '[FO4Edit](https://www.nexusmods.com/fallout4/mods/2737) **Quick Auto Clean**'
+    info: 'A cleaning guide is available [here](https://tes5edit.github.io/docs/5-mod-cleaning-and-error-checking.html#Apendix-B). It is strongly recommended not to use mods that contain **Deleted navmeshes** are they''re known to cause crashes. **Deleted navmeshes** must be corrected manually.'
```
```suggestion
    info: 'A cleaning guide is available [here](https://tes5edit.github.io/docs/5-mod-cleaning-and-error-checking.html#Apendix-B). It is strongly recommended not to use mods that contain **Deleted navmeshes** as they''re known to cause crashes. **Deleted navmeshes** must be corrected manually.'
```

Freso commented on masterlist.yaml:
```diff
@@ -394,43 +400,61 @@ plugins:
-        util: 'FO4Edit v3.2.33'
-        itm: 45
+      - <<: *quickClean
+        crc: 0xD69027EA # v1.10.114.0
```
I generally prefer to have the version on the line above the construct, possibly with explanatory notes. Ie.:
```yaml
      # v1.10.114.0, before cleaning
      - <<: *quickClean
        crc: 0xD69027EA
        …
      # v1.10.114.0, after initial cleaning(?)
      - <<: *quickClean
        crc: 0x1954F90B
        …
```

Just a comment though. Not something I'll hold up the PR over. :)

MacSplody replied:
I updated them anyway.

Freso commented on masterlist.yaml:
```diff
@@ -394,43 +400,61 @@ plugins:
+    clean:
+      - crc: 0xBD2A3927 # v1.10.114.0
+        util: 'FO4Edit v3.2.84 EXPERIMENTAL'
+        nav: 1
```
I don't think it's right to show `dirty` plugins as `clean`, even if there's nothing also that users (in most cases) can/should do about it. They're still dirty and may cause CTDs etc.

I just noticed that this is for the DLCs. Hm. I'm still not overly fond about this. DLCs are still dirty, even if they're as clean as they're likely to get.

MacSplody replied:
Okay I will leave `DLCs` with any `nav` remaining as blank with no `clean` or `reqManuaalFix` message.

Freso reviewed (changes requested):
It generally looks good. I have a few in-line comments directly to the code, with only real direct (and extremely minor) change request.

However, I'd like to see you clean up the commit messages message a bit. E.g., the first and third lines of https://github.com/loot/fallout4/pull/88/commits/840c648c5e21ed23a9d25eee987a636ebe38c8c7 are the same, basically repeating the commit "title" in the commit "body". The lists also seem a bit confusing to me, but maybe that's GitHub formatting them awkwardly (e.g., ignoring single line breaks?), I'll have to check locally to verify.

Freso committed:
Corrections for pull request

Fix Language used in reqManaulFix message.
Remove Clean entries with Deleted nav.
Update Version Comments.

MacSplody commented:
I think the commit message got messed up when I squashed a few commits before merging.

Ortham commented:
As a more general comment, is adding metadata generated using experimental versions of xEdit a good idea? By including it, are we not suggesting that users should use experimental versions when they should probably stick with proper releases?

MacSplody commented:
It's a fair point, I never really thought about it. I have used it extensively though & it has been very stable.
The [Beta](https://github.com/TES5Edit/TES5Edit/releases/tag/xedit-3.3.0) is available now if that would be better to display.

MacSplody commented:
Will open a new pull for the latest update.
